tests/test_jetmet_tools.py

```diff
@@ -837,9 +837,9 @@ def test_corrected_jets_factory(optimization_enabled):
             **{name: evaluator[name] for name in jec_stack_names[5:6]}
         )
 
-        print(dak.necessary_columns(jets.eta))
+        print(dak.report_necessary_columns(jets.eta))
```

Comment:
@lgray note that `report_necessary_columns` reports whatever the IO layer thinks a "column" is. It doesn't necessarily mean "dotted field" right now, rather I envisage it talking about the concept of a "key" or "column" that can be read from the IO source. 

@douglasdavis this is something I didn't fully think about when we made `necessary_columns` shim `report_necessary_columns`.

The difference will be that for remapped forms, the `report_necessary_columns` (and by proxy, `necessary_columns`) will report the underlying ROOT file keys, not the remapped keys.

Reply:
It therefore might mean that we should remove that alias for `necessary_columns`.

Reply:
I actually prefer the new behavior since users will probably expect to get a list of columns in the file they're reading! But seeing the remapped keys is nice for debugging.

Reply:
Can we make it so that we get a version or two of dask-awkward before `dak.necessary_columns` is gone and give migration instructions? I know a few end-users that use that function.

Reply:
@lgray I'm not 100% clear if we're on the same page. 

`report_necessary_columns` is a new function, so I'm comfortable with its semantics deviating from `ak.necessary_columns`. 

`ak.necessary_columns` is currently an alias for this new function, but for uproot this wouldn't be a non-breaking change; if previously it would report `ROOT_KEY.record.foo.bar`, now it would report `ROOT_KEY`. Is this OK? I don't see it being that useful if it breaks existing users.

Reply:
OK I suppose for my most typical use case I never see keys with `ROOT_KEY.record.foo.bar` formatting, it's really only ever `ROOT_KEY` and any structure is up in the form being remapped to. So they appear the same to me.